workers/docker-worker/src/lib/task.js

```diff
@@ -68,9 +68,15 @@ function buildStateHandlers(task, monitor) {
     throw new Error(`${diff.join()} ${diff.length > 1 ? 'are' : 'is'} not part of valid features`);
   }
 
+  diff = _.keys(featureFlags).filter(x => !runtime.features[x].enabled);
+
+  if (diff.length) {
+    throw new Error(`${diff.join()} ${diff.length > 1 ? 'are' : 'is'} not enabled on this worker`);
+  }
```

Comment:
I think the failure mode here isn't great (though I haven't tested it), as live log is not enabled at this point, and won't be if this check fails. I'm not sure of a good way around that without adding special handling for livelog, which I suspect is tricky.

Reply:
Sounds like a decent followup to file.  One idea might be to record this somewhere and report it as a warningwhen beginning the livelog.

Reply:
The issues is that livelog is itself a feature, and if this check fails, it won't be started. I suspect the fix is to special case livelog, so that it gets started even if this check fails.

Reply:
Ah, I remember now that "live log" in this case means "any logging at all", rather than just "the ability to see logs while the task is executing."  That's definitely a bummer.

Another option might be to throw an invalid-payload error here.

Anyway, I still think it's fine to leave this as a followup.  If someone does create a task that would have failed with a log, then at worst we dig through papertrail and see this message.

Reply:
#3341